cgi-bin/LJ/Talk.pm

```diff
@@ -1603,7 +1606,14 @@ sub talkform {
             ljuser => $remote->ljuser_display,
             openid_identity => $remote->openid_identity,
 
-            # allowed => ,
+            allowed => !$journalu->has_banned($remote) && (
```

Comment:
Feels like a further cleanup would be to pull a lot of this logic out to the user object, instead of having these many-fold conditionals in the middle of the talkform rendering. Then we could add tests around them and such...

Reply:
Huh. Should this all be rolled into the `does_not_allow_comments_from` method, or a new one?

Reply:
I would invert it probably, I always find it easier to reason about positive sounding conditions with if statements. Since most of the time we'd be writing something like,

```perl
return $bad_request->( 403 ) unless $u->allow_comments_from( $remote );
```

etc. FWIW, I don't think this is necessary for this patch, I'm just commenting for future work. I don't want to block merging the work already done!

Reply:
also I said if statements and then used an unless, go me